// Created on Feb 5.

syntax = "proto3";

package afs;

// Interface exported by the server.
service FileServer {
  // Open resp
  rpc Open(OpenReq) returns (OpenResp) {}
  
 /* // 
  rpc Read(ReadReq) returns (ReadResp) {}

 // 
  rpc Close(ReadReq) returns (ReadResp) {}

 // 
  rpc GetAttr(ReadReq) returns (ReadResp) {}
*/

  // File operations.
  rpc PutFile(PutFileReq) returns (PutFileResp) {} 
  rpc Delete(DeleteReq) returns (DeleteResp) {}
}

message OpenReq {
  string path = 1;
}

message OpenResp {
  int32 err = 1;
}

// TODO: make it streaming application.
message PutFileReq {
  string path = 1;

  // File content.
  bytes contents = 2;
}

message PutFileResp {
  int32 err = 1;
}

message DeleteReq {
  string path = 1;
}

message DeleteResp {
  int32 err = 1;
}
